prow/jenkins/controller.go

```diff
@@ -99,13 +99,23 @@ func NewController(kc *kube.Client, jc *Client, ghc *github.Client, logger *logr
 	}
 }
 
+func (c *Controller) config() config.Controller {
```

Comment:
I don't think this function works correctly if the deprecated `JenkinsOperator` config is used and `--label-selector` is specified. In that case `cfg.LabelSelectorString` will be "", but `c.selector` will be non empty so the correct config will not be found in the `JenkinsOperators` array.

Reply:
Fixed and added test cases.